validator/rules/executor.go

```diff
@@ -1,40 +1,38 @@
 package rules
 
 import (
-	"encoding/json"
 	"fmt"
-	"log"
 	"os"
 
 	"github.com/aerogear/charmil/validator"
 	"github.com/spf13/cobra"
 )
 
+// Rules is an interface that is implemented
+// by every rule defined in rules package
 type Rules interface {
-	Validate() []validator.ValidationError
+	Validate(cmd *cobra.Command) []validator.ValidationError
 }
 
 // RuleConfig is the struct that stores
 // configuration of rules
 type RuleConfig struct {
-	Verbose bool
-	Length
-	MustExist
+	Rules []Rules
 }
 
-// ExecuteRules executes all the rules
-// according to the RuleConfig provided
-func (config *RuleConfig) ExecuteRules(cmd *cobra.Command) []validator.ValidationError {
+// ExecuteRulesInternal executes all the rules
+// provided by ruleConfig
+func (ruleConfig *RuleConfig) ExecuteRulesInternal(cmd *cobra.Command, userValidatorConfig *ValidatorConfig) []validator.ValidationError {
```

Comment:
Do you dissagree with what I have proposed? 
In what I have proposed config should not have any method (not to mention internal)

Reply:
@wtrocki addressed comments